supervisor/addons/model.py

```diff
@@ -585,6 +588,10 @@ def uninstall(self) -> Awaitable[None]:
 
     def update(self) -> Awaitable[None]:
         """Update this add-on."""
+        if self.state in [AddonState.STARTING, AddonState.STOPPING]:
+            raise AddonsError(
+                f"Add-on is {self.state }, this is not a valid state for updates"
+            )
```

Comment:
We should move that to addon only. They are wrong place from me. I planned to remove that in the PR which we going away from central add-on data and make the objects more robust. The model object is just the base of the add-on config which we share with the StoreAddon and Addon. Currently it's better with latest revision which split the store out of the add-on but we need future work on that, also for the function to install different version of an add-on.